### PR TITLE
added section in gotchas about permutation

### LINKED ISSUE #20388: PermutationGroup fails to detect contention
Here are the steps to reproduce it:

    from sympy.combinatorics import Cycle
    from sympy.combinatorics.perm_groups import PermutationGroup

    perm_group = PermutationGroup(Cycle(2, 3)(4, 5), Cycle(1, 2, 3, 4, 5))

    [Cycle(perm) for perm in perm_group.elements]
    >>> [
            ...
            Cycle(1, 2, 3),
            ...
            Cycle(2, 5, 3),
            ...
        ]

    perm_group.contains(Permutation(Cycle(2, 5, 3)))
    >>> True

    perm_group.contains(Permutation(Cycle(1, 2, 3)))
    >>> False

### PR DESCRIPTION
**Pull request by wsdookadr**
#### References to other Issues or PRs
Fixes #20388 

#### Brief description of what is fixed or changed
Added a section to the permutation docs about containment in permutation groups.

* combinatorics
  * Added a section to the permutation docs about containment in permutation groups.

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
I don't think this really belongs in gotchas which is mostly about high-level confusions that users often have. I don't think that anyone would look there for information about permutations. It would make more sense for something about this to go into the docs for permutation if it isn't already there.

**Comment by wsdookadr** [comment]
> It would make more sense for something about this to go into the docs for permutation if it isn't already there.

Ok, I've placed it in `sympy/combinatorics/permutations.py`

File: sympy/combinatorics/permutations.py [hunk 1]
```diff
@@ -788,6 +788,54 @@ class Permutation(Atom):
...
+    permutation is being extended to 5 symbols by using a singleton,
+    and in the case of ``a3`` it's extended through the constructor
+    argument ``size=6``.
+
+    There is another way to do this, which is to tell the ``contains``
+    method that the number of symbols the group is on doesn't need to
+    match perfectly the number of symbols for the permutation, and that
+    can be done via the ``strict`` argument to the ``contains`` method,
+    and SymPy will try to extend the permutation on its own and then
+    perform the containment check.
```
Comment (oscarbenjamin) [inline sympy/combinatorics/permutations.py:837]: It would be good to have a doctest example showing how to use `strict` here.

Comment (wsdookadr) [reply sympy/combinatorics/permutations.py:839]: Ok, I've moved the doctest example closer to where it's being described

**Comment by oscarbenjamin** [comment]
Okay, this looks good. The tests haven't finished yet but +1 to merge.

The release note isn't needed.

**Comment by wsdookadr** [comment]
> Okay, this looks good. The tests haven't finished yet but +1 to merge.
> 
> The release note isn't needed.

Just tried to remove the release note but @sympy-bot fails a check if I do not include it.

**Comment by asmeurer** [comment]
This looks good, although I don't understand why strict=False isn't the default. I'm actually unclear why it's necessary in the group theory module to distinguish the same permutation in different groups. Couldn't we have S_1 ⊆ S_2 ⊆ S_3 ⊆ ...?

**Comment by asmeurer** [comment]
> Just tried to remove the release note but @sympy-bot fails a check if I do not include it.

You have to write `NO ENTRY` to have no release notes entry (see https://github.com/sympy/sympy/wiki/Writing-Release-Notes). But it's also fine to have an entry for this.

**Pull request merged**